File: app/src/main/java/com/cxwl/shawn/wuzhishan/decision/activity/WarningDetailActivity.java
package com.cxwl.shawn.wuzhishan.decision.activity;

import android.content.Context;
import android.os.Bundle;
import android.text.TextUtils;
import android.view.View;
import android.view.View.OnClickListener;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.cxwl.shawn.wuzhishan.decision.R;
import com.cxwl.shawn.wuzhishan.decision.dto.WarningDto;

/**
 * 预警详情
 */
public class WarningDetailActivity extends BaseActivity implements OnClickListener {
	
	private Context mContext = null;
	private LinearLayout llBack = null;
	private TextView tvTitle,tvName,tvTime,tvIntro;

	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		setContentView(R.layout.activity_warning_detail);
		mContext = this;
		initWidget();
	}
	
	/**
	 * 初始化控件
	 */
	private void initWidget() {
		llBack = findViewById(R.id.llBack);
		llBack.setOnClickListener(this);
		tvTitle = findViewById(R.id.tvTitle);
		tvTitle.setText("预警详情");
		tvName = findViewById(R.id.tvName);
		tvTime = findViewById(R.id.tvTime);
		tvIntro = findViewById(R.id.tvIntro);

		WarningDto data = getIntent().getParcelableExtra("data");
		if (data != null) {
			if (!TextUtils.isEmpty(data.name)) {
				tvName.setText(data.name);
			}
			if (!TextUtils.isEmpty(data.time)) {
				tvTime.setText(data.time);
			}
			if (!TextUtils.isEmpty(data.content)) {
				tvIntro.setText(data.content);
			}
		}
	}
	
	@Override
	public void onClick(View v) {
		switch (v.getId()) {
		case R.id.llBack:
			finish();
			break;

		default:
			break;
		}
	}

}

File: app/src/main/java/com/cxwl/shawn/wuzhishan/decision/activity/WarningDetailActivity.kt
package com.cxwl.shawn.wuzhishan.decision.activity

import android.os.Bundle
import android.text.TextUtils
import android.view.View
import android.view.View.OnClickListener
import com.cxwl.shawn.wuzhishan.decision.R
import com.cxwl.shawn.wuzhishan.decision.dto.WarningDto
import kotlinx.android.synthetic.main.activity_warning_detail.*
import kotlinx.android.synthetic.main.layout_title.*

/**
 * 预警详情
 */
class WarningDetailActivity : BaseActivity(), OnClickListener {
	
	override fun onCreate(savedInstanceState: Bundle?) {
		super.onCreate(savedInstanceState)
		setContentView(R.layout.activity_warning_detail)
		initWidget()
	}
	
	/**
	 * 初始化控件
	 */
	private fun initWidget() {
		llBack.setOnClickListener(this)
		tvTitle.text = "预警详情"

		val data : WarningDto? = intent.getParcelableExtra<WarningDto>("data")
		if (data != null) {
			if (!TextUtils.isEmpty(data.name)) {
				tvName.text = data.name
			}
			if (!TextUtils.isEmpty(data.time)) {
				tvTime.text = data.time
			}
			if (!TextUtils.isEmpty(data.content)) {
				tvIntro.text = data.content
			}
		}
	}

	override fun onClick(view: View?) {
		when(view!!.id) {
			R.id.llBack -> finish()
		}
	}

}
